Extension/src/LanguageServer/configurations.ts

```diff
@@ -1044,11 +1039,18 @@ export class CppProperties {
 
             // Get the text of the current configuration.
             let curText: string = document.getText();
+
+            // Replace all \<escape character> with \\<character>, except for \"
+            // Otherwise, the JSON.parse result will have the \<escape character> missing.
+            let configurationsText: string = util.escapeForSquiggles(curText);
+            let configurations: ConfigurationJson = JSON.parse(configurationsText);
+            let currentConfiguration: Configuration = configurations.configurations[this.CurrentConfigurationIndex];
+
             let curTextStartOffset: number = 0;
-            if (!this.CurrentConfiguration.name) {
+            if (!currentConfiguration.name) {
```

Comment:
I'm pretty sure `this.CurrentConfiguration` is used as a safety net that is guaranteed not to be undefined.  Please check before changing this.